include/url.h

```diff
@@ -15,22 +15,51 @@ namespace Url
         UrlParseException(const std::string& message) : std::logic_error(message) {}
     };
 
+    struct CharacterClass
+    {
+        CharacterClass(const std::string& chars) : chars_(chars), map_(256, false)
+        {
+            for (auto it = chars_.begin(); it != chars_.end(); ++it)
+            {
+                map_[static_cast<size_t>(*it)] = true;
+            }
+        }
+
+        bool operator()(char c) const
```

Comment:
I would probably use `contains` as a method name rather than operator overloading for this meaning, but I don't feel strongly about it.